backend/src/main/java/gov/cdc/usds/simplereport/config/AuthorizationConfiguration.java

```diff
@@ -0,0 +1,45 @@
+package gov.cdc.usds.simplereport.config;
+
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.config.annotation.method.configuration.EnableGlobalMethodSecurity;
+
+import gov.cdc.usds.simplereport.config.authorization.UserAuthorizationVerifier;
+import gov.cdc.usds.simplereport.config.simplereport.AdminEmailList;
+import gov.cdc.usds.simplereport.service.OrganizationService;
+import gov.cdc.usds.simplereport.service.model.IdentitySupplier;
+
+/**
+ * Configuration (mostly static) for method-level or object-level security in
+ * the application (as opposed to request-level security, which lives in
+ * {@link SecurityConfiguration}).
+ */
+@Configuration
+@EnableGlobalMethodSecurity(prePostEnabled = true)
+public class AuthorizationConfiguration {
+
+    public static final String AUTHORIZER_BEAN = "simpleReportAuthVerifier";
+
+    @Bean(AUTHORIZER_BEAN)
+    public UserAuthorizationVerifier getVerifier(AdminEmailList admins, IdentitySupplier supplier,
+            OrganizationService orgService) {
+        return new UserAuthorizationVerifier(admins, supplier, orgService);
+    }
+
+    /**
+     * Apply this annotation if the method should only be called by site-wide
+     * administrative users ("superusers").
+     */
+    @Retention(RUNTIME)
+    @Target(METHOD)
+    @PreAuthorize("@" + AUTHORIZER_BEAN + ".userHasSiteAdminRole()")
```

Comment:
instead of an annotation just for admins, is there a way we can make an annotation that takes, as arguments, the roles the caller must have in order to call that method? you know spring magic better than i do, but that would seem more extensible.

Reply:
I think this is a great idea, in that I had the same idea myself and keep banging my head against the "surely this can't be impossible?!" wall. Sadly, there is no way to do it using simple `@PreAuthorize` meta-annotations—we'd have to write some custom code to plug into a slightly deeper layer of Spring Security, and this does not seem worth the time.

Reply:
You sure? :) 
https://docs.spring.io/spring-security/site/docs/4.1.x/reference/html/el-access.html#access-control-using-preauthorize-and-postauthorize

Reply:
Yes, pretty sure. As far as I can tell (and as far as any of the other people who have asked StackOverflow this question have been able to tell) you can look at method arguments with SPeL, but not annotation attributes.